pykechain/enums.py

```diff
@@ -909,3 +909,40 @@ class WidgetTitleValue(Enum):
     DEFAULT = 'Default'
     NO_TITLE = 'No title'
     CUSTOM_TITLE = 'Custom title'
+
+
+class NotificationStatus(Enum):
+    """
+    Options to retrieve a Notification based on its status.
+
```

Comment:
maybe show the status flow here. I guess there is somewhere an ascii drawing on the flow let met get it

```transition_matrix = {
            NotificationStatus.DRAFT: [NotificationStatus.READY],
            NotificationStatus.READY: [NotificationStatus.PROCESSING],
            NotificationStatus.PROCESSING: [NotificationStatus.COMPLETED, NotificationStatus.FAILED],
            NotificationStatus.COMPLETED: [NotificationStatus.ARCHIVED],
            NotificationStatus.FAILED: [NotificationStatus.ARCHIVED],
            NotificationStatus.ARCHIVED: []
        }
```

```
    normal lifecycle:

    - DRAFT, when a message is first saved to the backend and the status is still in draft.
        next states: READY
    - READY: when the message is ready for processing, it is complete and is to be processed
        next states: PROCESSING
    - PROCESSING: when the message is in the process of being send out
        next states: COMPLETED, FAILED
    - COMPLETED: when the message is successfully sent out
        next states: ARCHIVED
    - FAILED: when the message is not successfully sent out
        next states: ARCHIVED
    - ARCHIVED: when the message is archives and waiting for its deletion against a certain retention policy
        next states: None
```

Reply:
Gedaan